src/EditorFeatures/Core/Implementation/FindReferences/DocumentLocation.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.FindReferences
+{
+    /// <summary>
+    /// Represents a <see cref="TextSpan"/> location in a <see cref="Document"/>.
+    /// </summary>
+    internal struct DocumentLocation : IEquatable<DocumentLocation>, IComparable<DocumentLocation>
+    {
+        public Document Document { get; }
+        public TextSpan SourceSpan { get; }
+
+        public DocumentLocation(Document document, TextSpan sourceSpan)
+        {
+            Document = document;
+            SourceSpan = sourceSpan;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals((DocumentLocation)obj);
+        }
+
+        public bool Equals(DocumentLocation obj)
+        {
+            return this.CompareTo(obj) == 0;
+        }
+
+        public static bool operator ==(DocumentLocation d1, DocumentLocation d2)
+        {
+            return d1.Equals(d2);
```

Comment:
This doesn't handle d1 being null.  I usually `return object.Equals(d1, d2);` for this, because it does handle it, though it goes through the extra casting back through...